nowcasting_dataset/manager.py

```diff
@@ -0,0 +1,384 @@
+    ) -> None:
+        """Initialise DataSources specified in the InputData configuration.
+
+        For each key in each DataSource's configuration object, the string `<data_source_name>_`
+        is removed from the key before passing to the DataSource constructor.  This allows us to
+        have verbose field names in the configuration YAML files, whilst also using standard
+        constructor arguments for DataSources.
+        """
+        for data_source_name in names_of_selected_data_sources:
+            logger.debug(f"Creating {data_source_name} DataSource object.")
+            config_for_data_source = getattr(self.config.input_data, data_source_name)
+            if config_for_data_source is None:
+                logger.info(f"No configuration found for {data_source_name}.")
+                continue
+            config_for_data_source = config_for_data_source.dict()
+
+            # Strip `<data_source_name>_` from the config option field names.
+            config_for_data_source = nd_utils.remove_regex_pattern_from_keys(
+                config_for_data_source, pattern_to_remove=f"^{data_source_name}_"
+            )
+
+            data_source_class = MAP_DATA_SOURCE_NAME_TO_CLASS[data_source_name]
+            try:
+                data_source = data_source_class(**config_for_data_source)
+            except Exception:
+                logger.exception(f"Exception whilst instantiating {data_source_name}!")
+                raise
+            self.data_sources[data_source_name] = data_source
+
+        # Set data_source_which_defines_geospatial_locations:
+        try:
+            self.data_source_which_defines_geospatial_locations = self.data_sources[
+                self.config.input_data.data_source_which_defines_geospatial_locations
+            ]
+        except KeyError:
+            logger.error(
+                "input_data.data_source_which_defines_geospatial_locations="
+                f"{self.config.input_data.data_source_which_defines_geospatial_locations}"
+                " is not a member of the DataSources, so cannot set"
+                " self.data_source_which_defines_geospatial_locations!"
+                f" The available DataSources are: {list(self.data_sources.keys())}"
+            )
+        else:
+            logger.info(
+                f"DataSource `{data_source_name}` set as"
+                " data_source_which_defines_geospatial_locations."
+            )
+
+    def create_files_specifying_spatial_and_temporal_locations_of_each_example_if_necessary(
+        self,
+    ) -> None:
+        """Creates CSV files specifying the locations of each example if those files don't exist yet.
+
+        Creates one file per split, in this location:
+
+        `<output_data.filepath> / <split_name> / spatial_and_temporal_locations_of_each_example.csv`
+
+        Works on any compute environment.
+        """
+        if self._locations_csv_file_exists():
+            return
```

Comment:
add logger statement in here? just to say the csv files already exists - might help with some confusion later down the line

Reply:
Good point!  Done!